docker-compose.example.yml

```diff
@@ -26,32 +41,55 @@ services:
       - ./localstack_bootstrap.sh:/etc/localstack/init/ready.d/localstack_bootstrap.sh
     networks:
       - standardnotes_self_hosted
+    healthcheck:
+      test: curl -s http://localhost:4566/_localstack/health
+      timeout: 10s
+      interval: 5s
+      start_period: 60s
 
   db:
     image: mysql:8
     container_name: db_self_hosted
+    user: "${UID}:${GID}"
     environment:
+      - LANG=C.UTF-8
       - MYSQL_DATABASE=standard_notes_db
       - MYSQL_USER=std_notes_user
       - MYSQL_ROOT_PASSWORD=changeme123
       - MYSQL_PASSWORD=changeme123
+      - MYSQL_INITDB_CHARSET=utf8mb4
     expose:
       - 3306
-    restart: unless-stopped
-    command: --default-authentication-plugin=mysql_native_password --character-set-server=utf8mb4 --collation-server=utf8mb4_general_ci
     volumes:
       - ./data/mysql:/var/lib/mysql
       - ./data/import:/docker-entrypoint-initdb.d
     networks:
       - standardnotes_self_hosted
+    healthcheck:
+      test: [
+        "CMD", 
+        "mysqladmin",
+        "ping", 
+        "-h", "localhost",
+        '-u', 'root', 
+        '-p$$MYSQL_ROOT_PASSWORD'
+      ]
+      timeout: 20s
+      retries: 10
+    restart: unless-stopped
 
   cache:
-    image: redis:6.0-alpine
+    image: redis:7
```

Comment:
we are using `6` on production so it has to stay at that. Also why use the Debian based image instead of Alpine?

Reply:
Alpine have errors with dns resolutions and the difference in size and performance don't pay the random errors, I already using with 7 and looks is working, but if you think that have breaking changes, let me know to revert to 6.

Reply:
I’m not saying it won’t work but until we migrate our production infra to 7 then we are staying at 6 as a validation during e2e tests